tests/main/main.tf

```
@@ -9,17 +9,29 @@ module "main" {
   repo_endpoint = local.repo_endpoint
   repos = [
     {
-      yum_prefix       = local.yum_prefix
+      # Test using cloudfront endpoint for main repo
+      salt_s3_bucket   = local.salt_s3_bucket
       salt_s3_endpoint = local.salt_s3_endpoint
       salt_versions    = local.salt_versions
       repo_prefix      = local.repo_prefix
+      yum_prefix       = local.yum_prefix
     },
     {
-      repo_prefix      = local.repo_prefix_archive
+      # Test using cloudfront endpoint for archive repo
+      salt_s3_bucket   = local.salt_s3_bucket
       salt_s3_endpoint = local.salt_s3_endpoint_archive
       salt_versions    = local.salt_versions_archive
+      repo_prefix      = local.repo_prefix_archive
       yum_prefix       = local.yum_prefix
     },
+    {
+      # Test using the underlying archive bucket directly, instead of cloudfront
+      salt_s3_bucket   = "archive-repo-saltstack-com"
+      salt_s3_endpoint = "s3.us-west-2.amazonaws.com"
+      salt_versions    = local.salt_versions_archive
+      repo_prefix      = "repo/underlying_bucket/"
```

If I understand it correctly, should this be using `local.repo_prefix_archive`?
```suggestion
      repo_prefix      = "${local.repo_prefix_archive}/underlying_bucket/"
```

---

They way the logic works, it runs a separate sync once for each entry here. Each sync will _delete_ content that it is not managing. So these prefixes cannot overlap. `repo/archive/underlying_bucket/` would overlap with `repo/archive/`, so that won't work out as expected.

---

Ah.  Yeah, that makes sense.